pkg/agent/controller/egress/egress_controller.go

```diff
@@ -241,13 +263,11 @@ func (c *EgressController) addEgress(obj interface{}) {
 }
 
 // updateEgress processes Egress UPDATE events.
-func (c *EgressController) updateEgress(old, cur interface{}) {
-	oldEgress := old.(*crdv1a2.Egress)
+func (c *EgressController) updateEgress(_, cur interface{}) {
 	curEgress := cur.(*crdv1a2.Egress)
-	// Ignore handling the Egress Status change if Egress IP already has been assigned on current node.
-	if curEgress.Status.EgressNode == c.nodeName && oldEgress.GetGeneration() == curEgress.GetGeneration() {
-		return
-	}
+	// We need to sync the Egress once even if its spec doesn't change as a Node's EgressIP capacity may be exceeded
+	// when multiple Egresses were processed in parallel and were assigned to the same Node.
+	// Re-sync after status change could correct the assignment eventually.
```

Comment:
I assume it's not very convenient to write a unit test for this edge case?

Reply:
Yes, it would need at least two goroutines run in parallel and one goroutine must select the node before the other goroutine updates egress status, which is hard to control. The results may vary on different testbeds and it's hard to predict expected calls.